### PR TITLE
os/bluestore: make mutex_debug_impl default constructible

### PR DESCRIPTION
**Pull request by aclamk**
common/mutex: make mutex_debug_impl DefaultConstructible
os/bluestore: give mutex a name.

Signed-off-by: A.Kupczyk@redhat.com

### CODE REVIEW

**Comment by aclamk** [comment]
This supersedes #24528.

File: src/os/bluestore/BitmapFreelistManager.h [hunk 1]
```diff
@@ -9,14 +9,15 @@
...
+#include "common/ceph_mutex.h"
 #include "include/buffer.h"
 #include "kv/KeyValueDB.h"
 
 class BitmapFreelistManager : public FreelistManager {
   std::string meta_prefix, bitmap_prefix;
   KeyValueDB *kvdb;
   std::shared_ptr<KeyValueDB::MergeOperator> merge_op;
-  std::mutex lock;
+  ceph::mutex lock = {ceph::make_mutex("BitmapFreelistManager::lock")};
```
Comment (tchaikov) [inline src/os/bluestore/BitmapFreelistManager.h]: i think we can either use brace or equal for the default initializer:
```c++
ceph::mutex lock = ceph::make_mutex("BitmapFreelistManager::lock");
```

or 

```c++
ceph::mutex lock{ceph::make_mutex("BitmapFreelistManager::lock")};
```
but no need to use both of them at the same time.

File: src/common/mutex_debug.h [hunk 2]
```diff
@@ -93,6 +93,10 @@ class mutex_debug_impl : public mutex_debugging_base
   static constexpr bool recursive = Recursive;
 
   // Mutex concept is DefaultConstructible
+  mutex_debug_impl(bool bt = false)
+    : mutex_debugging_base("unnamed", bt) {
+    _init();
+  }
```
Comment (cbodley) [inline src/common/mutex_debug.h]: is this change just motivated by convenience? i worry that making the name optional just makes the debug mutexes less useful in general

Comment (liewegas) [reply src/common/mutex_debug.h]: yeah i'd rather require the names

**Pull request closed without merging**